be/src/storage/task/engine_publish_version_task.cpp

```diff
@@ -51,35 +50,17 @@ Status EnginePublishVersionTask::finish() {
         return Status::NotFound(fmt::format("Not found tablet to publish_version. tablet_id: {}, txn_id: {}",
                                             _tablet_info.tablet_id, _transaction_id));
     }
-
-    Status st = Status::OK();
-    if (tablet->keys_type() == KeysType::PRIMARY_KEYS) {
-        VLOG(1) << "UpdateManager::on_rowset_published tablet:" << tablet->tablet_id()
-                << " rowset: " << _rowset->rowset_id().to_string() << " version: " << _version;
-        st = StorageEngine::instance()->txn_manager()->publish_txn2(_transaction_id, _partition_id, tablet, _version);
-    } else {
-        st = StorageEngine::instance()->txn_manager()->publish_txn(_partition_id, tablet, _transaction_id,
-                                                                   Version{_version, _version});
-    }
+    auto st = StorageEngine::instance()->txn_manager()->publish_txn(_partition_id, tablet, _transaction_id, _version,
```

Comment:
why remove publish_txn2?

Reply:
publish_txn & publish_txn2 has lot of code in common, so I merge these two functions into one